src/hotspot/share/runtime/frame.hpp

```diff
@@ -440,6 +440,7 @@ class frame {
   void interpreter_frame_print_on(outputStream* st) const;
   void print_on_error(outputStream* st, char* buf, int buflen, bool verbose = false) const;
   static void print_C_frame(outputStream* st, char* buf, int buflen, address pc);
+  static frame next_frame(frame fr, Thread* t); // For native stack walking
```

Comment:
A thought: Does it make more sense to have next\_frame implemented in vmError or nativeStackPrinter and made available through their respective headers rather than as a member of the frame class? Just thinking out loud, I'm fine with it as is

Reply:
I figured the only class that should know that much about how to walk frames should the frame class. In addition it would look for either of the other classes to export that kind of API for other classes to use.

Thanks for re-reviewing. Of course it needs another review now I fixed the indentation issue.

Reply:
Ah, I was talking about having it as a top level extern method in either nativeStackPrinter or vmError, and declared in their headers for other code to use, not as a member of any of their classes, but I guess it doesn't matter. Thanks for your patience

I don't think my review would've mattered anyway even if you hadn't changed anything since I'm not a Reviewer, but for the sake of it I'll re\-approve :)